src/NHibernate/Util/EnumerableExt.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Collections;
+
+namespace NHibernate.Util
+{
+	internal static class EnumerableExt
+	{
+		internal static int Count(this IEnumerable source)
```

Comment:
This method takes precedence over standard `Enumerable.Count<T>` in some cases.

Reply:
Fixed (forced push).